### PR TITLE
Change Limit defition to refelct v1beta3

### PR DESCRIPTION
**Pull request by jsmolar**
Resolves https://github.com/Kuadrant/testsuite/issues/575

- Changes Limit object to reflect the current Limit api
- Swaps currently used Rules from RLP to CEL expressions and predicates

### CODE REVIEW

File: testsuite/kuadrant/policy/__init__.py [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+    """Dataclass that references CEL predicate e.g. auth.identity.anonymous == 'true'"""
+
+    predicate: str
+
+
+@dataclass
+class CelExpression:
+    """Dataclass that references CEL expression"""
+
+    expression: str
```
Comment (averevki) [inline testsuite/kuadrant/policy/__init__.py:20]: I'd leave these 2 as a string. Also, they should be under RateLimit module if they are only used there

**Review by averevki — APPROVED** [review]

File: testsuite/kuadrant/policy/__init__.py (hunk 1 above)
Comment (jsmolar) [reply testsuite/kuadrant/policy/__init__.py:20]: what do you mean by leaving them as string?

**Review by martinhesko — APPROVED** [review]

File: testsuite/kuadrant/policy/__init__.py (hunk 1 above)
Comment (averevki) [reply testsuite/kuadrant/policy/__init__.py:20]: I don't see the reason to use dataclasses here

Comment (jsmolar) [reply testsuite/kuadrant/policy/__init__.py:20]:
```
      when:
        - predicate: "request.method == 'POST'" 
```

This is predicate in action. For this you would need to specify "predicate" every time

**Pull request merged**